### PR TITLE
Outline text for resolution

### PR DESCRIPTION
**Pull request by jeromekelleher**
Here's a start on resolution - any thoughts @grahamgower?

### CODE REVIEW

File: docs/specification.md [hunk 1]
```diff
@@ -505,23 +505,49 @@ for definitive information on the structural properties of the
...
+
+Resolution happens in a set of steps in a defined order, which we
+describe here. Please consult the
+{ref}`reference implementation<sec_spec_reference_implementation>`
+for more detailed information.
+
+A demes model is resolved by first resolving each
+{ref}`deme<sec_spec_hdm_resolution_deme>` in an order that
+guarantees that a deme is resolved *after* all of its ancestors
+have been resolved. Subsequently,
```
Comment (grahamgower) [inline docs/specification.md:538]: I think this should say the following things:
* demes are resolved in the order in which they appear in the input
* a deme can only be resolved if its ancestor demes have already been resolved
* the parser MUST raise an error if a deme is encountered that has unresolved ancestors
* this guarantees that for any valid input file, the demes are sorted topologically (ancestors are listed before their descendants)

File: docs/specification.md [hunk 2]
```diff
@@ -505,23 +505,49 @@ for definitive information on the structural properties of the
...
 :::
 
 start_time
-: The most ancient time at which the deme exists, in ``time_units``
-  before the present. Demes with no ancestors are root demes and must
-  have an infinite ``start_time``. Otherwise, the ``start_time`` must
-  correspond with the interval of existence
-  for each of the deme's ``ancestors``. I.e. the ``start_time`` must
-  be within the half-open interval ``(deme.start_time, deme.end_time]``
-  for each deme in ``ancestors``.
```
Comment (grahamgower) [inline docs/specification.md:522]: I guess this is all stated in the MDM section?

File: docs/specification.md [hunk 3]
```diff
@@ -531,17 +557,18 @@ start_time
    - If the deme has no ancestors, the ``start_time`` shall be
      infinitely far into the past. I.e. the ``start_time`` shall
      have the value ``infinity``.
+   - Otherwise the ``start_time`` cannot be determined and an
+     error should be raised.
```
Comment (grahamgower) [inline docs/specification.md:561]: MUST be raised

**Review by grahamgower — CHANGES_REQUESTED** [review]
I think we should more explicitly outline the deme resolution order here.

File: docs/specification.md (hunk 1 above)
Comment (jeromekelleher) [reply docs/specification.md:538]: done

**Comment by jeromekelleher** [comment]
Made those changes @grahamgower. I'll follow up with any other changes separately, let's get this much merged.

**Pull request merged**